app/bod/management/commands/bod_migrate.py

```diff
@@ -0,0 +1,104 @@
+from bod.models import BodContactOrganisation
+from provider.models import Provider
+
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+
+class Command(BaseCommand):
+    help = "Migrates data from a BOD"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.messages = []
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--clear",
+            action="store_true",
+            help="Delete existing objects before importing",
+        )
+        parser.add_argument(
+            "--dry-run",
+            action="store_true",
+            help="Dry run, abort transaction in the end",
+        )
+        parser.add_argument(
+            "--verbose",
+            action="store_true",
+            help="Show more output",
+        )
+
+    def clear(self):
+        cleared, _ = Provider.objects.all().delete()
+        if cleared:
+            self.messages.append(f"{cleared} provider(s) cleared")
+
+    def import_providers(self, verbose):
+        added = 0
+        updated = 0
+        removed = 0
+        processed = set()
+
+        for organization in BodContactOrganisation.objects.all():
+            legacy_id = organization.pk_contactorganisation_id
+            processed.add(legacy_id)
+            provider, created = Provider.objects.get_or_create(_legacy_id=legacy_id)
+            changed = False
+            for provider_attribute, organization_attribute in (
+                ('name_de', 'name_de'),
+                ('name_de', 'name_de'),
+                ('name_en', 'name_en'),
+                ('name_fr', 'name_fr'),
+                ('name_it', 'name_it'),
+                ('name_rm', 'name_rm'),
+                ('acronym_de', 'abkuerzung_de'),
+                ('acronym_fr', 'abkuerzung_fr'),
+                ('acronym_en', 'abkuerzung_en'),
+                ('acronym_it', 'abkuerzung_it'),
+                ('acronym_rm', 'abkuerzung_rm'),
+            ):
+                old = getattr(provider, provider_attribute)
+                new = getattr(organization, organization_attribute)
+                if old != new:
+                    changed = True
+                    setattr(provider, provider_attribute, new)
+                    if verbose and not created:
+                        self.stdout.write(
+                            f"Changed provider {provider.id} ({legacy_id}) "
+                            f"{provider_attribute} from '{old}' to '{new}'"
+                        )
+            if changed and not created:
+                updated += 1
+            if created:
+                added += 1
+                if verbose:
+                    self.stdout.write(f"Added provider '{organization.name_en}'")
+
+            provider.save()
+
+        removed, _ = Provider.objects.filter(_legacy_id__isnull=False).exclude(
```

Comment:
Is removing providers a case? Chris is mentionning that he would use the mechanism in prod as well. Could be the case that we already have some providers in there, now at some point we want to migrate the old ones too, then we want to keep the manually entered ones.
(Could be solved with a flag)

Reply:
Yes, that's why I only remove providers which came intially from the BOD (`_legacy_id__isnull=False`).

Reply:
@msom @schtibe for some of the models it could even get worse, potential sources of data are bod, geocat, and this service itself. We might make the source of the data explicit at some point with a field `data_origin` or `data_master` or sth. (and values `bod`, `geocat`, `api`, `admin` or sth like this). Doesn't need to be in this PR though